java/runtime/src/main/java/io/ray/runtime/RayNativeRuntime.java

```diff
@@ -174,7 +174,7 @@ public void start() {
 
   @Override
   public void shutdown() {
-    Lock writeLock = shutdownLock.readLock();
+    Lock writeLock = shutdownLock.writeLock();
```

Comment:
Two questions:
1. Will `shutdown ` be called by multithreading?
2. Why there is no problem with UT?

Reply:
1. `shutdown` won't be called concurrently, but the lock is also used in `NativeObjectStore`. When an object is GCed, the object will be unregistered from core worker. Since GC runs in a separate thread, we need to make sure that core worker is available when `NativeObjectStore` is accessing core worker in the GC thread.

2. There was some occasional failures in ReferenceCountingTest, but since we've added retry recently (like Python tests), it's unlikely to fail 3 times in a test session.